opendps/func_cv.c

```diff
@@ -285,7 +285,7 @@ static void cv_tick(void)
     uint16_t i_out_raw, v_in_raw, v_out_raw;
     hw_get_adc_values(&i_out_raw, &v_in_raw, &v_out_raw);
     /** Continously update max voltage output value */
-    cv_voltage.max = (float) pwrctl_calc_vin(v_in_raw) / 1.1f + 0.5f; /** Max output voltage is 1.1 * Vin */
+    cv_voltage.max = (float) pwrctl_calc_vin(v_in_raw) / VIN_VOUT_RATIO + 0.5f; /** Max output voltage is 1.1 * Vin */
```

Comment:
Oh, and one more thing. If we have devised the ratio, why add another 0.5V? Just curious...

Reply:
When we typecast from a float to a unsigned integer the value after the decimal point is lost i.e. 4.7 will become 4. This is obviously wrong as we would rather it gets rounded to 5. By adding 0.5 to it we will always round correctly

This only works for positive numbers however as cv_voltage.max is unsigned we are safe to make this assumption

Reply:
I made some similar changes [on my repo over here](https://github.com/Xenoamor/opendps/commit/adbd69af11f66b9347dc27b3d2e50535e2ae5bfc)

Reply:
Right! Could you please add that to the comment which I now see should read "Max output voltage is Vin / 1.1"

Reply:
Done and done. Well spotted!
`cv_voltage.max` is actually a signed int but `pwrctl_calc_vin()` returns a unsigned int so we should never have a negative float value and this rounding method will still work